### ISSUE TITLE
Pulling significantly less Pokemon

### ISSUE THREAD

**Issue by 2-m** [AHAAAAAAA/PokemonGo-Map#1463]
The newest versions of both the develop branch and the master branch are pulling significantly less Pokemon than before. I had to use @mchristopher 's latest release to find a Persian, which wasn't showing up on my map despite how many times I had it rescan

**Comment by almasmaris** [comment]
I'm also trying the newest stable version on master branch. It show less pokemon than previous version. My brother still use the oldest version so i can compare how many pokemon appear on both version.

**Comment by Yivoe** [comment]
**New Version**

![new version](https://cloud.githubusercontent.com/assets/20549349/17076591/39da121a-5073-11e6-9c3f-7a692f724df9.png)

**Comment by Dantergy** [comment]
Yeah. I've compared with the previous version, and the new version show less pokemon :/

**Comment by lrobson555** [comment]
@Yivoe Which branch and commit and you running in the old version?

**Comment by shme14** [comment]
@Yivoe Can you post a screenshot with the scanning area turned on?

**Comment by GamingMaster2000** [comment]
I'm using the latest develop commit and I have found changing lat_gap_meters and lng_gap_meters in search.py to 100 and 50 respectively picks up most, if not all, Pokemon (where as some were missing with the default values). My scanning areas said they were scanned but they weren't finding anything until I lowered the values.

I don't know what the optimum values are. I literally just picked those numbers because they were nice round numbers and lower than the default but it seemed to work :p Of course by lowering the gap between steps I needed to increase the number of steps to cover the same area.

**Comment by Yivoe** [comment]
@lrobson555  gonna have to tell me where to find that version #.  I know I downloaded it a couple days ago.

@shme14  Here you go:

![newverwithscan](https://cloud.githubusercontent.com/assets/20549349/17076663/4041554e-5075-11e6-928a-03b0dcca1772.png)

**Comment by hhu94** [comment]
@Yivoe go into each of the directories and type git log. You should see a list of commit numbers. Can you give us the latest number of the two versions you are comparing?

**Comment by 2-m** [comment]
@GamingMaster2000 

@mchristopher's latest 0.1.0 release: http://i.imgur.com/Mjyw5bt.png

Develop branch: http://i.imgur.com/kD2PD5U.png

No 'mons filtered, this is all I see

**Comment by hhu94** [comment]
@2-m What do you mean by 0.1.0 release?

**Comment by 2-m** [comment]
@hhu94 https://github.com/mchristopher/PokemonGo-DesktopMap/releases

**Comment by hhu94** [comment]
This is what I see with the current master branch:

![2016-07-23-00 39 45](https://cloud.githubusercontent.com/assets/7145349/17076708/04e50ee8-506e-11e6-9b88-7626f6507bbd.png)

**Comment by marauder37** [comment]
I'm struggling to replicate this. I have the two versions running side by side and logging what they find. `runserver.py found 652 of 735, 88%`. And that's without bothering to make the search areas the same size.

**Comment by marauder37** [comment]
This is my Utah map after about 60s of runtime: https://snag.gy/Dkjtgb.jpg

**Comment by Yivoe** [comment]
@hhu94  

 **Most recent thing on the new version:**

commit 5d321405b287acdccb1e43c44563e021d300e5fb
Author: mab <github@e.marauder.net.au>
Date:   Sat Jul 23 13:26:16 2016 +1000

    Change default display to no pokestops, no scan bubbles (#1425)


Most Recent thing on the Old Version:

**Comment by Yivoe** [comment]
@marauder37   That is how I want mine to look!  Unfortunately, as you can see from the screenshots above, this new version is giving me almost no pokemon

**Comment by marauder37** [comment]
@yivoe you can see from the map title that mine is the new version also. Really struggling to replicate your map. Are you using a Google account or PTC?

**Comment by Yivoe** [comment]
google account.  could that make a difference?

**Comment by marauder37** [comment]
@yivoe try some of those suggestions... smaller lat_gap_meters as per talentedmrripley, also try `--threads 1` which someone else said helps

**Comment by hhu94** [comment]
@mchristopher's latest 0.1.0 release on the left, latest master branch on the right:
![2016-07-23-00 52 52](https://cloud.githubusercontent.com/assets/7145349/17076778/cca57a84-506f-11e6-905c-9eecdc00d85d.png)

**Comment by Yivoe** [comment]
@marauder37  I missed the threads comment.  What exactly am I doing there?  And I just tried the smaller lat_gap meters, it did increase the pokemon being shown, but it requires me to increase my -st by a lot to cover the same areas, and it takes a lot longer.  Not really feasible at some point.

@hhu94 Did you modify anything in either of those before you ran them?

**Comment by hhu94** [comment]
@Yivoe I did not modify anything, they're clean copies.

**Comment by 2-m** [comment]
@marauder37 Trying --threads 1 now, I'll report back with results

**Comment by pickett2k** [comment]
changing the long and lat in search.py has made a significant improvement for me also runnning side by side

**Comment by hhu94** [comment]
@Yivoe Your old version on the left and your new version on the right, I checked out the two commits you say you have.

![2016-07-23-01 04 29](https://cloud.githubusercontent.com/assets/7145349/17076825/837c4534-5071-11e6-9181-c420b2d4e600.png)

There must be something else causing your problems :/

**Comment by almasmaris** [comment]
Current master branch. I downloaded a while ago
![pokemon](https://cloud.githubusercontent.com/assets/5098639/17076834/21bcb3ee-50e7-11e6-91fa-3880648a15ec.PNG)

Old version (on my brother laptop)
![pokemon](https://cloud.githubusercontent.com/assets/5098639/17076838/503d94ae-50e7-11e6-8527-d16b065b7b2d.JPG)

**Comment by Yivoe** [comment]
@hhu94 Thanks for checking that.  I just can't figure out whats causing it.  I just downloaded @mchristopher 's latest 0.1.0, and I got awesome results.  Everything seemed to work well.  But I downloaded the Master and got the problems I've been showing you.

Made no changes to either of them.

Here it is side by side (running at the same time):

![sidebyside](https://cloud.githubusercontent.com/assets/20549349/17076846/8606d8ce-507a-11e6-80d8-4d90928c6a6a.png)

**Comment by ApocUSB190** [comment]
@Yivoe Can you go into /pogom/models.py and comment out the following lines:
```
    if pokestops:
        log.info("Upserting {} pokestops".format(len(pokestops)))
        bulk_upsert(Pokestop, pokestops)
```

**Comment by keaza** [comment]
i got rid of pokestops and gyms, i have found this has made quite a diffrence.

**Comment by Yivoe** [comment]
@ApocUSB190  I have tried that before, and I just tried it again.  It still looks the same as the comparison I posted above...

**Comment by ApocUSB190** [comment]
Interesting, sorry to waste your time.  I cannot replicate the bug you are experiencing :(

**Comment by hhu94** [comment]
@Yivoe I'm using PTC.

**Comment by Yivoe** [comment]
Not a waste, thank you for trying to help me.  I was excited for the update because it looked like it wouldn't crash after ~30min anymore (so I could leave it running at my apt and go out), but I've spent my night trying to get it to work at all....

**Comment by Yivoe** [comment]
@hhu94 Side note:  Shit, I just saw your comment in the other thread.  I missed that snorlax  :/  Part of the reason I need to get this working for me.

**Comment by marauder37** [comment]
@yivoe I'm going to leave http://pogo.mab.cat:5010/ running for a while, I hope it brings you some joy

**Comment by JZ6** [comment]
for everyone posting screen shots, please remember to use two separate accounts while testing, as using the same account will cause one of them to run a lot slower.
Please provide console screenshots as well

**Comment by klaphekje78** [comment]
I have done some testing, and it seems the multiethreading part on default is nog showing most pokemon (default is 5) im running on 2 or 3 and the map is showing much more pokemon (command line '-t 2'

**Comment by schadis** [comment]
i changed to -t 1 
seems to show again that amount of pokemons as before

**Comment by Yivoe** [comment]
@klaphekje78 Thank you!  I don't know if this is going to be perfect and get every pokemon, but it is looking like it is doing it.

@schadis Changing it to 1 from 2 made it even better

Is it safe to assume this will be consistent?

**Comment by Yivoe** [comment]
And to add, with -t 1, it seems to be going faster and getting more pokemon than I was getting with @mchristopher 's latest 0.1.0

**Comment by 2-m** [comment]
Do --threads and -t not do the same thing?

**Comment by marauder37** [comment]
@2-m yes, I just prefer longoptions

**Comment by klaphekje78** [comment]
lowering the threading paramater will make the map build slower. but more accurate. dont know why the pokemons are not showing. if you look at the logs you will see the same amount of pokemon seen in with different values of threading. maybe it has something to do with the speed of writing it to the db. 

im not a developer so cant comment on that part.

**Comment by almasmaris** [comment]
@marauder37 which version are you using?

**Comment by marauder37** [comment]
I'm using the latest develop and it's fine on OSX and AWS EC2 so I would like people who are having problems to say whether they're using Windows or not

**Comment by JakeCooper** [comment]
Chatted with a couple people, and all of them that were able to reproduce this issue were on Windows (Myself included).

**Comment by marauder37** [comment]
I'm on the develop branch as it happens but they're very similar as of a few hours ago. I am happy to switch branches if somebody thinks it's relevant

**Comment by almasmaris** [comment]
@marauder37 i'm trying develop branch and it still show less pokemon. I'm on windows.

**Comment by marauder37** [comment]
It is with deep regret that I'm now booting a Windows VM to try it there

**Comment by marauder37** [comment]
Also I have enjoyed the wide variety of different places that people on this thread moved my instance's map centre to, but that party is over for the time being

**Comment by noaqzz** [comment]
using latest dev branch too, and its showing less pokemon as before.
With -t 1      I had 99% of the pokemons.

**Comment by D3nn3** [comment]
![vs](https://cloud.githubusercontent.com/assets/20516965/17077184/12e801ea-50c7-11e6-90b0-f5481c617008.jpg)

Windows 10 with two different google accounts and pulled the master branch about 1,5h ago.
Left -t 15 | Right -t 1

**Comment by cNyhuus** [comment]
I have just changed from -st 12 -t 10 to -st 5 and -t
and then added more locations to fill the same area. I seem to get a lot more too, with the small amount of theads, but multiple instances insted.

**Comment by almasmaris** [comment]
I'm using -t 1 and it show more pokemon and same like old version.

**Comment by almasmaris** [comment]
@marauder37 can you tell me how to upload it to server and can be access with http?

**Comment by marauder37** [comment]
Go to https://github.com/AHAAAAAAA/PokemonGo-Map and pound on the purple "Deploy to Heroku" button and play along, I'm told that works well. Deploying to AWS is out of scope for this discussion

**Comment by jlroettger** [comment]
Or just change the host to 0.0.0.0 on your computer and open up the port with your router :-)

**Comment by Yivoe** [comment]
I ran away from this thread for a bit.  After that -t 1 fix got me going a dragonite popped up like 2 minutes later...

Anyways, @marauder37  I'm also on windows.  Problems were in both windows 10 and 7 (separate machines).  I also used the develop and the Master on both machines, same problem on both.

-t 1 is the only thing that has helped so far, but I would love a more sustainable fix, as this one causes the map to fill fairly slowly.

**Comment by marauder37** [comment]
Yes I can replicate the problem in Windows, I'll see if I can find out the cause before I need therapy

**Comment by Yivoe** [comment]
Doing the lords work @marauder37   Thanks for all your help.

**Comment by codengine** [comment]
Just a guess: May it be that Niantic has some kind of anti-hammering / DDOS protection, which lets some requests just drop if the requests come in too frequently?

**Comment by hhu94** [comment]
@codengine That cannot be because that wouldn't discriminate operating systems.

**Comment by marauder37** [comment]
I think I have a fix for this, I'll get some testing for it

**Comment by marauder37** [comment]
https://github.com/AHAAAAAAA/PokemonGo-Map/pull/1486 feedback appreciated

**Comment by marauder37** [comment]
Nope, spoke too soon

**Comment by Yivoe** [comment]
I was just in the middle of commenting.  It hadn't fixed it for me either.  Still had to use -t 1 to get em all

**Comment by slonce** [comment]
While there is no solution?

**Comment by marauder37** [comment]
```
2016-07-23 21:16:49,461 [     search] [   INFO] Step 2 has the lock
2016-07-23 21:16:49,976 [     search] [   INFO] Step 2 releases the lock
2016-07-23 21:16:49,976 [     search] [   INFO] Step 1 has the lock
2016-07-23 21:16:50,506 [     search] [   INFO] Step 1 releases the lock
2016-07-23 21:16:50,506 [     search] [   INFO] Step 8 has the lock
2016-07-23 21:16:51,022 [     search] [   INFO] Step 5 has the lock
2016-07-23 21:16:51,022 [     search] [   INFO] Step 8 releases the lock
2016-07-23 21:16:51,520 [     search] [   INFO] Completed  0.37% of scan.
2016-07-23 21:16:51,520 [     search] [   INFO] Completed  0.74% of scan.
2016-07-23 21:16:51,536 [     search] [   INFO] Step 3 has the lock
2016-07-23 21:16:51,536 [     search] [   INFO] Step 5 releases the lock
2016-07-23 21:16:52,052 [     search] [   INFO] Step 3 releases the lock
```

I can't tell whether Windows thread locking really is that broken, or there's a problem with output, but someone good with Windows Python threads needs to take a look

**Comment by mstampfli** [comment]
i had the same issue since today morning, the -t 1 solved the issue. i tested on also on unix, centos 7, manual installation. i had no issues there.. so yeah maybe its windows related :/

**Comment by Josephvb10** [comment]
Still no fix?

**Comment by Maxxxel** [comment]
Experiencing same problem on Windows 10 64 bit, PTC account.

**Comment by austinxw** [comment]
experiencing same problem on EC2, ubuntu. --thread 1 fixed.

**Comment by stsfin** [comment]
Having the same problem on Ubutu and PTC with raspberry pi, what I found is adding a longer scan delay helps and lesser steps. When its closer to being a "real walk"-speed and single threaded it gives better results. So it seems that Niantic has added some hammer protection

**Comment by slimjim2234** [comment]
👍 
same issue here

**Comment by MysticalOS** [comment]
Late to ticket but I'm noticing it's using too big of a scan radius. are you account for fact they nerfed pop radius from 100m down to 70m? That you need to use a 70m scan size now not 100 or you'll miss a lot.

**Comment by alexandre1984** [comment]
When I run the runserver.py, I got this error.
usage: runserver.py [-h] [-a AUTH_SERVICE] [-u USERNAME] [-p PASSWORD]
                    [-l LOCATION] [-st STEP_LIMIT] [-sd SCAN_DELAY]
                    [-ld LOGIN_DELAY] [-dc] [-H HOST] [-P PORT] [-L LOCALE]
                    [-c] [-d] [-m] [-ns] [-os] [-nsc] [-fl] -k GMAPS_KEY [-C]
                    [-D DB] [-cd] [-np] [-ng] [-nk] [--db-type DB_TYPE]
                    [--db-name DB_NAME] [--db-user DB_USER]
                    [--db-pass DB_PASS] [--db-host DB_HOST]
                    [--db-max_connections DB_MAX_CONNECTIONS]
                    [-wh [WEBHOOKS [WEBHOOKS ...]]]
runserver.py: error: unrecognized arguments: -t 1

I'm on windows 10.

Any help?

**Comment by sawaldt** [comment]
@alexandre1984 Looks like the way threading is handled was changed and the above argument removed in #2956
